my_enums.rb

```diff
@@ -0,0 +1,118 @@
+# frozen_string_literal: true
+
+# rubocop:disable Metrics/AbcSize, Metrics/CyclomaticComplexity, Metrics/MethodLength, Metrics/PerceivedComplexity
+
+module Enumerable #:nodoc: all
+  def my_each
+    length.times do |n|
+      current = self[n]
+      yield(current)
+    end
+  end
+
+  def my_each_with_index
+    length.times do |n|
+      index = n
+      current = self[n]
+      yield(index, current)
+    end
+  end
+
+  def my_select
+    emp_arr = []
+    my_each do |n|
+      emp_arr << n if yield(n)
+    end
+    emp_arr
+  end
+
+  def my_all?
+    emp_arr = []
+    my_each do |n|
+      emp_arr << n if yield(n)
+    end
+    emp_arr == self
+  end
+
+  def my_any?
+    tester = false
+    my_each do |n|
+      tester = true if yield(n)
+      break if tester
+    end
+    tester
+  end
+
+  def my_none?
+    tester = true
+    my_each do |n|
+      tester = false if yield(n)
+      break if tester == false
+    end
+    tester
+  end
```

Comment:
- [ ] enumerables my_none? when no block or argument is given returns true only if none of the collection members is true
- [ ] enumerables my_none? when a class is passed as an argument returns true if none of the collection is a member of such class
- [ ] enumerables my_none? when a Regex is passed as an argument returns true only if none of the collection matches the Regex  
- [ ] enumerables my_none? when a pattern other than Regex or a Class is given returns true only if none of the collection matches the pattern